percona/controller/pgcluster/finalizer.go

```diff
@@ -150,6 +153,61 @@ func (r *PGClusterReconciler) stopExternalWatchers(ctx context.Context, cr *v2.P
 	return nil
 }
 
+func (r *PGClusterReconciler) deleteBackups(ctx context.Context, cr *v2.PerconaPGCluster) error {
+	log := logging.FromContext(ctx)
+
+	podList := &corev1.PodList{}
+	err := r.Client.List(ctx, podList, &client.ListOptions{
+		Namespace: cr.Namespace,
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			naming.LabelPerconaInstance:  cr.Name,
+			naming.LabelPerconaComponent: "pg",
+		}),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	if len(podList.Items) == 0 {
+		return errors.New("no pods found")
+	}
+
+	log.Info("Deleting backups from all the repos configured")
+
+	pod := podList.Items[0]
+
+	var stdout, stderr bytes.Buffer
+	pgBackrestCmd := "pgbackrest --stanza=db --log-level-console=info stop; " +
```

Comment:
should we run the second command if first one fails?

Reply:
[bbe958a](https://github.com/percona/percona-postgresql-operator/pull/989/commits/bbe958a6aadc539e29b51de6739f332ce82b6bb6)